src/test/js/shared/idam_helper.js

```diff
@@ -408,34 +388,48 @@ class IdamHelper extends Helper {
         return result;
     }
 
-    extractUrlFromBody(emailResponse) {
-        if (emailResponse) {
-            const regex = "(https.+)"
-            const url = emailResponse.body.match(regex);
-            if (url[0]) {
-                return url[0].replace(/https:\/\/idam-web-public[^\/]+/i, TestData.WEB_PUBLIC_URL).replace(")", "");
+    async extractUrlFromNotifyEmail(searchEmail) {
+        let emailResponse;
+        let url;
+        let i = 0;
+        while (i < maxRetries && !emailResponse) {
+            emailResponse = await this.getEmailFromNotify(searchEmail);
+            if (emailResponse) {
+                const regex = "(https.+)";
+                const urlMatch = emailResponse.body.match(regex);
+                if (urlMatch[0]) {
+                    url = urlMatch[0].replace(/https:\/\/idam-web-public[^\/]+/i, TestData.WEB_PUBLIC_URL).replace(")", "");
+                }
```

Comment:
maybe a one second sleep before retrying,  if we don't have an email response? Probably worth a log message as well.